lib/app/services/api.dart

```diff
@@ -44,6 +48,67 @@ class ApiService {
     }).toList();
   }
 
+  Future<List<Course>> listCourses() async {
+    var response = await network.get('courses?');
+
+    var body = json.decode(response.body);
+
+    return Future.wait((body['data'] as List<dynamic>).map((data) async {
+      data = data as Map<String, dynamic>;
+      return Course(
+          id: Id<Course>(data['_id']),
+          name: data['name'],
+          description: data['description'],
+          teachers: [
+            for (String id in data['teacherIds']) await getUser(Id<User>(id))
```

Comment:
By using a `Future.wait` here, we can fetch all the users in parallel instead of sequentially. If all of these need to be fetched over the network, that brings huge performance increasements.